dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/AbstractTimerTask.java

```diff
@@ -22,24 +22,24 @@
 import org.apache.dubbo.common.timer.TimerTask;
 import org.apache.dubbo.remoting.Channel;
 
-import java.util.Collection;
 import java.util.concurrent.TimeUnit;
 
 /**
  * AbstractTimerTask
  */
 public abstract class AbstractTimerTask implements TimerTask {
 
-    private final ChannelProvider channelProvider;
+
+    private final Channel channel;
 
     private final Long tick;
 
-    AbstractTimerTask(ChannelProvider channelProvider, Long tick) {
-        if (channelProvider == null || tick == null) {
+    AbstractTimerTask(Channel channel, Long tick) {
+        if (channel == null || tick == null) {
```

Comment:
I think putting both the null check together make it defficult to trace for which this exception has been raised.

I think we should refactor this to 

```
Assert.notNull(channel,"Channel can't be null');
Assert.notNull(tick,"Tick can't be null");
```

Assert is under package of org.apache.dubbo.common.utils. They way it works is, if the given first argument is null then it throws IllegalArgumentException with messaged getting passed here as second argument. 

What do you say?